lookout/style/format/tests/test_quality_report.py

```diff
@@ -69,39 +71,88 @@ def __exit__(self, *args):
+            self.assertEqual(data["max_conf"], 0.9999756217002869)
+            self.assertEqual(data["min_conf"], 0.19736842811107635)
+            self.assertEqual(data["max_support"], 20528)
+            self.assertEqual(data["min_support"], 16)
+            lines = """|Min support|16|
+|Max support|20528|
+|Min confidence|0.19736842811107635|
+|Max confidence|0.9999756217002869|""".splitlines()
+            for line in lines:
+                self.assertIn(line, output)
+            num_rules, avg_len = _get_model_summary(output)
+            self.assertEqual(num_rules, 1269)
+            self.assertEqual(avg_len, 19.10401891252955)
 
     def test_eval(self):
         """Test on normal input."""
         input_pattern = os.path.join(self.jquery_dir, "**", "*")
         with Capturing() as output:
-            quality_report(input_pattern=input_pattern, bblfsh=self.bblfsh,
-                           language=self.language, model_path=self.model_path,
-                           config={"uast_break_check": False})
-        self.assertEqual(["# Quality report", "", "### Classification report:"], output[:3])
-        self.assertIn("### Rules summary:", output)
+            print_reports(input_pattern=input_pattern, bblfsh=self.bblfsh,
+                          language=self.language, model_path=self.model_path,
+                          config={"uast_break_check": False})
+        self.assertEqual([
+            "# Quality report for javascript / <unknown url> <unknown reference> <unknown commit>",
+            "",
+            "### Classification report"],
+            output[:3])
+        qcount = output.count(
+            "# Quality report for javascript / <unknown url> <unknown reference> <unknown commit>")
+        self.assertEqual(qcount, 14)
+        self.assertIn("### Summary", output)
+        self.assertIn("# Model report for <unknown url> <unknown reference> <unknown commit>",
+                      output)
+        self.assertGreater(len(output), 100)
+        self.assertIn("javascript", _get_json_data("\n".join(output)))
+
+    def test_eval_aggregate(self):
+        """Test on normal input, quality reports are aggregated."""
+        input_pattern = os.path.join(self.jquery_dir, "**", "*")
+        with Capturing() as output:
+            print_reports(input_pattern=input_pattern, bblfsh=self.bblfsh,
+                          language=self.language, model_path=self.model_path,
+                          config={"uast_break_check": False, "aggregate": True})
+        qcount = output.count(
+            "# Quality report for javascript / <unknown url> <unknown reference> <unknown commit>")
+        self.assertEqual(qcount, 1)
+        output = "\n".join(output)
+        output = output[:output.find("# Model report for <unknown url>")]
+        metrics = _get_precision_recall_f1_support(output)
+        self.assertEqual(
+            metrics, (0.7443157321054488, 0.7700059276822763, 0.7323340980410543, 3374))
```

Comment:
can this numbers be changed suddenly after some changes in the code?

Reply:
Yes, so this is a good indicator of such changes. Let's see if we have to change them frequently, leaving for now